app/assets/javascripts/components/autocomplete.js

```diff
@@ -4,18 +4,63 @@ var $autocompletes = document.querySelectorAll('[data-module="autocomplete"]')
 if ($autocompletes) {
   $autocompletes.forEach(function ($el) {
     var customAttributes = {}
-    if ($el.attributes['data-contextual-guidance']) {
+    var $select = $el.querySelector('select')
+    if ($select.attributes['data-contextual-guidance']) {
       customAttributes = {
-        'data-contextual-guidance': $el.attributes['data-contextual-guidance'].value
+        'data-contextual-guidance': $select.attributes['data-contextual-guidance'].value
       }
     }
 
     // disabled eslint because we can not control the name of the constructor (expected to be EnhanceSelectElement)
     new window.accessibleAutocomplete.enhanceSelectElement({ // eslint-disable-line no-new, new-cap
-      selectElement: $el,
+      selectElement: $select,
       minLength: 3,
       showNoOptionsFound: true,
       customAttributes: customAttributes
     })
   })
 }
+
+var $customTemplateAutocomplete = document.querySelector('[data-module="autocomplete-custom-template"]')
```

Comment:
I'm a little confused here, but hopefully just my misunderstanding. The name 'autocomplete-custom-template' made me think we would be passing an HTML template in somewhere. But it looks like the 'custom-template' just changes the HTML in a fixed way; therefore, could we call it 'autocomplete-with-hint' or similar, to express that it's specific look 'n feel?